<launch>

    <!-- input imu topic name -->
    <arg name="input_topic" default="/imu"/>

    <!-- input imu topic name -->
    <arg name="output_topic" default="/imu/data"/>

    <!--
         transform to this frame and publish tf
         (source frame will be used from incoming message)
     -->
    <arg name="target_frame" default="cog"/>

    <!-- define nice value of process (lower means higher priority) [-20;19] -->
    <!-- more infos: https://en.wikipedia.org/wiki/Nice_(Unix) -->
    <arg name="nice" default="nice -n -5"/>

    <!-- forward output to [screen|log] -->
    <arg name="output" default="screen"/>

    <!-- default parameter config file -->
    <arg name="config_file" default="$(find drive_ros_imu_filter)/config/cc2017_car.yaml" />

    <!-- apply transform -->
    <node name="imu_apply_transform" pkg="drive_ros_imu_filter"
          type="apply_transform" output="$(arg output)" launch-prefix="$(arg nice)">
        <remap from="~/imu_in" to="$(arg input_topic)"/>
        <remap from="~/imu_out" to="$(arg output_topic)"/>
        <param name="target_frame" type="str" value="$(arg target_frame)" />
    </node>

    <!-- broadcast transform -->
    <node name="imu_broadcast_transform" pkg="drive_ros_imu_filter"
          type="broadcast_transform" output="$(arg output)" launch-prefix="$(arg nice)">
        <remap from="~/imu_in" to="$(arg input_topic)"/>
        <param name="target_frame" type="str" value="$(arg target_frame)" />
        <rosparam command="load" file="$(arg config_file)"/>
    </node>

</launch>
